lib/puppet/type/package.rb

```diff
@@ -611,5 +611,63 @@ def refresh
         provider.reinstall
       end
     end
+
+    HOLD_DOC="Valid values are: \"yes\"/\"true\"/\"no\"/\"false\""
+    newproperty(:hold, :required_features => :holdable) do
+      desc <<-EOT
+        Set to true to tell Debian apt/Solaris pkg to hold the package version
+        
+        #{HOLD_DOC}
+        Default is "false". Hold can be specified with or without `ensure`,
+        if `ensure` is missing will default to "present".
+
+        Hold cannot be specified together with "purged", "absent" or "held"
+        values for `ensure`.
+      EOT
+      newvalues(:true, :false, :yes, :no)
+      munge do |value|
+        case value
+        when true, :true, 'true', :yes, 'yes'
+          :true
+        when false, :false, 'false', :no, 'no'
+          :false
+        else
+          raise ArgumentError, _("Invalid hold value %{value}. %{doc}") % { value: value.inspect, doc: HOLD_DOC}
+        end
+      end
+
+      def insync?(is)
+        (@should[0] == :true) == is
+      end
+
+      def should_to_s(value)
+        (value == :true).to_s
+      end
```

Comment:
Does `is_to_s` need to be overridden?

Reply:
without it will print something like: 
`Changing hold from "false" to true`